### ISSUE TITLE
Not all patron groups at Shibboleth-enabled library have credentials

### ISSUE THREAD

**Issue by gibsonjc** [CARLI/vufind#214]
UIS - The new VuFind update is super rad and a much needed improvement to the interface. I especially love the Shib integration but since that’s the only option people who don’t have Shib credentials, like friend groups, won’t be able to log in. Might I suggest implementing something like Springshare uses where you can give both options of logging in with either your VuFind account or Shib.

**Comment by gibsonjc** [comment]
I asked UIS: Regarding the UIS patron groups that do not have Shibboleth-level credentials on your campus, what functions do they need to be able to use in the catalog?  For example:
•         place local requests?  
•         place requests for materials from other I-Share libraries?  
•         be able to look up their account online for fines, etc.?
•         be able to renew books they have checked out?
•         other?

Answer: Patrons without Shibboleth level creds can do everything except request from I-Share libraries.

**Comment by gibsonjc** [comment]
UIC - 12/15/2017 open call

**Comment by dmcmpbll** [comment]
I asked UIU for input on this on 2/20/2017:
"Courtesy card holders  (LOCAL LOPRV, LOCAL HIPRV, LOCAL BUO) and  LIBRARYUSE cards (possibly ILLINETLIB ? ) aren’t able to claim NetIDs and set up passwords. There may be exceptions, but it would be a handful of the +/-400 cards that we issue throughout the year (looking back to FY13-14 through FY15/16."

This would mean that they would be unable to place local requests or manage their accounts through the OPAC.

**Comment by dmcmpbll** [comment]
UIU- remains interested in the affects of only offering the Institutional-affilliated login for such-enabled institutions; UIU has ~400 new cards each year where the patron will not have an institutional login account.

**Comment by cedelis** [comment]
I've put together a proof-of-concept on DEVEL for UIU: https://vufind4-devel.carli.illinois.edu/vf-uiu/

If you click on the "Login" link, you will automatically be taken to the campus site login page (Shibboleth) via some javascript "magic." You might notice the loading of intermediary page as it takes a fraction of a second for the javascript to take you to the landing (campus site) login page. I could probably do better with that (perhaps make the page hidden by default); but I think you get the point.

There is support for ILS-based login (Barcode/Last Name), too. However, in order to initiate this particular login, you will need to access the "hidden" URL directly, e.g., https://vufind4-devel.carli.illinois.edu/vf-uiu/MyResearch/UserLogin?auth_method=ILS

What do you think?

BTW, there is more than one way to solve this issue. Above, I have outlined a method of using a "hidden" URL. Another approach I could take is to create a completely separate VuFind site URL for this. I.e., give the local catalog its own unique base URL; e.g., instead of https://vufind4-devel.carli.illinois.edu/vf-uiu, we could use something like https://vufind4-devel.carli.illinois.edu/vf-uiu-ils  Similar, to the method I described above, this approach would also, in essence, make use of "hidden" URL (in that it wouldn't be directly integrated with the default local catalog). But, it would be "cleaner" behind-the-scenes (from a maintenance and configuration perspective).

**Comment by cedelis** [comment]
Ready for Testing.

**Comment by cedelis** [comment]
Shibboleth library example on TEST: https://i-share-test.carli.illinois.edu/vf-uiu/MyResearch/UserLogin?auth_method=ILS

**Comment by dmcmpbll** [comment]
> Shibboleth library example on TEST: https://i-share-test.carli.illinois.edu/vf-uiu/MyResearch/UserLogin?auth_method=ILS

I tested the link above as myself, and as a "LOCAL LOPRV" patron for UIU-test.

For myself, I was able to log in both with my campus credentials, and/or my barcode/name.
For the LOCAL LOPRV patron, I could log in with their barcode/name (since they have no credentials). 

It seemed to work great!

We may want to work a little on the wording on the screen to make it it more evident that these are both login options for the catalog, but the functionality seems to be working. I'll come up with a draft for wording that others can critique (and then we can see if it's possible to edit).

**Comment by gibsonjc** [comment]
@dmcmpbll and I discussed this a bit today and below are some questions to investigate.  We will likely need @eschell to comment on possible customization options:

1. Is the “Login/Institutional Login” wording in the upper left of the page customizable with the library’s terminology for what their campus calls logging in? 
Like for UIUC, could it say “Log in with your NetID and password”
And for Millikin, it could say: “Log in with your MyMillikin username and password”

2. Underneath or somewhere near the hopefully-customizable-Login-text link could there be a link to the separate auth_method=ILS page with text like, “If you do not have a NetID, click here” or “If you do not have a MyMillikin account, click here” and the auth_method=ILS page that they land on could have text that contains a description of what the NetID/MyMillikin is, and then “enter your Community Borrower barcode" (or whatever they are called at that institution) and Last Name below. If that doesn’t work, contact library here”

3. If at all possible we want to avoid there being a screen like is currently available 
https://i-share-test.carli.illinois.edu/vf-uiu/MyResearch/Home (when you haven't logged in) 
that has both options on it because credentialed users will likely **not** read the explanatory text and just type in the boxes 
![screen shot 2018-02-07 at 8 18 54 am](https://user-images.githubusercontent.com/17598357/35921253-cd8aefa6-0bdf-11e8-953c-c7ee38d22277.png)

**Comment by cedelis** [comment]
@dmcmpbll and @gibsonjc : I've refactored some of the Auth templates ( a6cf2a5516696805300c15d5b94b9eaae8e1f3c8 ), which solved the last bullet point below. This refactoring was also done to hopefully make it easier for @eschell to accomplish the remaining [ I've included comments/suggestions to @eschell below ]: 

   [ @eschell , I see that you've already pulled down this refactoring into your dev region, so you may begin working on this at any time. ]

1. Is the “Login/Institutional Login” wording in the upper left of the page customizable with the library’s terminology for what their campus calls logging in?
Like for UIUC, could it say “Log in with your NetID and password”
And for Millikin, it could say: “Log in with your MyMillikin username and password”  

    [ @eschell , the templates you may want to edit: themes/carli/templates/Auth/Shibboleth/login.phtml - if you wish to edit around the Shibboleth link  **or** if you want to edit the page that contains both the Shibboleth and MultiILS templates (the one with the text, "Please choose a login method") , you can override bootstrap3's implementation of themes/bootstrap3/templates/Auth/ChoiceAuth/login.phtml by copying that into carli's theme directory themes/carli/templates/Auth/ChoiceAuth/ and then editing carli's version ]

1. Underneath or somewhere near the hopefully-customizable-Login-text link could there be a link to the separate auth_method=ILS page with text like, “If you do not have a NetID, click here” or “If you do not have a MyMillikin account, click here” and the auth_method=ILS page that they land on could have text that contains a description of what the NetID/MyMillikin is, and then “enter your Community Borrower barcode" (or whatever they are called at that institution) and Last Name below. If that doesn’t work, contact library here”  

    [ @eschell , you will probably want to edit the Shibboleth template, themes/carli/templates/Auth/Shibboleth/login.phtml , and also the MultiILS template(s), themes/carli/templates/Auth/MultiILS/login.phtml and/or themes/carli/templates/Auth/MultiILS/loginfields.phtml  ,  that is referenced by the former ]

1. ~~If at all possible we want to avoid there being a screen like is currently available
https://i-share-test.carli.illinois.edu/vf-uiu/MyResearch/Home (when you haven't logged in)
that has both options on it because credentialed users will likely not read the explanatory text and just type in the boxes~~


@ed , An example UR you can use to bring up the page with both Auth choices (Shibboleth,MultiILS) in your dev environment: https://vufind4-mockup.carli.illinois.edu/vf-uiu/MyResearch/UserLogin?auth_method=ILS

**Comment by gibsonjc** [comment]
Customization templates/variables have been set up were pushed to production on 4/6.  

Customization form now available at https://www.carli.illinois.edu/newvufind-shibboleth-login-customization.  

Today I emailed contacts from ILC, LFC, MIL, NCC, NIU, NPU, SIC, UIC, UIS, UIU to let them know the options and customization form are now available.  This form will be sent to any library that enables single sign-on in future.  Closing.
